index.js

```diff
@@ -44,6 +44,14 @@ var isArray = Array.isArray;
 // interdependence between different functions.
 var conjure;
 
+// a special hasOwnProperty function so we can override the built-in one
+
+var hasOwnProperty = Object.prototype.hasOwnProperty;
+
+function hop(obj, propName) {
```

Comment:
I actually took this naming [from another project](https://github.com/isaacs/node-lru-cache/blob/ba1abfab6443d4a9f9fe5a3c277a216ca13e75ba/lib/lru-cache.js#L10-L12) where I saw it being used and kinda liked it. Short and sweet, and clear that it is *not* the built-in Object.prototype.hasOwnProperty function. I think that distinction is important (so Tome.hasOwnProperty is a no-go for me). I will await feedback from others on this one. Would like to know what they think.